### PR TITLE
Support for a `qualityGateStatus` output variable

### PR DESCRIPTION
**Pull request by aldass**
Added support for a `qualityGateStatus` output variable. From the action, this variable is set to a value of `PASSED`, `WARN` or `FAILED`. It can optionally be used as follows below.

The motivation for this change is the need to be able to programmatically use the _SonarQube Quality Gate_ value in other parts of a complex GitHub workflow via the standard GitHub Workflow syntax conventions.

```yaml
- name: SonarQube Quality Gate check
  id: sqQualityGateCk # id can be any uniqe value throught the yml file
  uses: sonarsource/sonarqube-quality-gate-action@master
  with:
    scanMetadataReportFile: target/sonar/report-task.txt
- name: "Example show SonarQube Quality Gate Status value"
  run: echo "The Status is ${{ steps.sqQualityGateCk.outputs.qualityGateStatus }}"
```

### CODE REVIEW

**Comment by kubamarchwicki** [comment]
:rocket: that's a super helpful change when one looks for ways to customize the pipelines and sonar notifications (beyond the PR decorations). We use sonarcloud for some of our private projects and this simple change will greatly enhance and simplify some of our build pipelines.

File: script/check-quality-gate.sh [hunk 1]
```diff
@@ -37,12 +37,16 @@ qualityGateUrl="${serverUrl}/api/qualitygates/project_status?analysisId=${analys
 qualityGateStatus="$(curl --silent --fail --show-error --user "${SONAR_TOKEN}": "${qualityGateUrl}" | jq -r '.projectStatus.status')"
 
 if [[ ${qualityGateStatus} == "OK" ]];then
+   echo '::set-output name=qualityGateStatus::PASSED'
    success "Quality Gate has PASSED."
 elif [[ ${qualityGateStatus} == "WARN" ]];then
+   echo '::set-output name=qualityGateStatus::WARN'
    warn "Warnings on Quality Gate."
 elif [[ ${qualityGateStatus} == "ERROR" ]];then
+   echo '::set-output name=qualityGateStatus::FAILED'
```
Comment (wouter-admiraal-sonarsource) [inline script/check-quality-gate.sh:46]:
```suggestion
   echo '::set-output name=quality-gate-status::FAILED'
```

File: script/check-quality-gate.sh [hunk 2]
```diff
@@ -37,12 +37,16 @@ qualityGateUrl="${serverUrl}/api/qualitygates/project_status?analysisId=${analys
...
    fail "Quality Gate has FAILED."
 else
+   echo '::set-output name=qualityGateStatus::FAILED'
```
Comment (wouter-admiraal-sonarsource) [inline script/check-quality-gate.sh:49]:
```suggestion
   echo '::set-output name=quality-gate-status::FAILED'
```

File: README.md [hunk 3]
```diff
@@ -34,36 +34,58 @@ jobs:
...
+The `qualityGateStatus` output variable from the action is set to a value of `PASSED`, `WARN` or `FAILED`. It can optionally be used as follows:
+
 ```yaml
 - name: SonarQube Quality Gate check
+  id: sqQualityGateCk # id can be any uniqe value throught the yml file
   uses: sonarsource/sonarqube-quality-gate-action@master
   with:
     scanMetadataReportFile: target/sonar/report-task.txt
+- name: "Example show SonarQube Quality Gate Status value"
+  run: echo "The Status is ${{ steps.sqQualityGateCk.outputs.qualityGateStatus }}"
```
Comment (wouter-admiraal-sonarsource) [inline README.md:88]: Let's drop this. It's already explained above, no point in repeating it.

**Review by wouter-admiraal-sonarsource — CHANGES_REQUESTED** [review]
Hi @aldass!

Thank you very much for your contribution! I've added some change requests, mainly nitpicks. And I would align with the common practice of naming action output variables using lowercased and hyphened words, rather than camelcase.

Finally, could you add 3 extra assertions in the unit tests ([example here](https://github.com/SonarSource/sonarqube-quality-gate-action/blob/master/test/check-quality-gate-test.bats#L68)) 🙏 ? That way, we can prevent accidental regressions.

Please let me know if you need any help with this.

P.S. I've created a ticket on our side to track this: [SQQGGHA-2](https://jira.sonarsource.com/browse/SQQGGHA-2)

**Comment by wouter-admiraal-sonarsource** [comment]
Hi again. I've taken the liberty to fork your fork, and make the suggested updates over there 🙂. You can follow the new PR [here](https://github.com/SonarSource/sonarqube-quality-gate-action/pull/17). I'll close this one. Thanks again for your contribution!

**Pull request closed without merging**